### DISCUSSION TITLE
How can I use localePath inside Cypress tests?

### DISCUSSION THREAD

**Discussion by theolavaux** [nuxt-modules/i18n#1880] [General]
I'm migrating an app from Nuxt 2 to Nuxt 3 and I'm currently refactoring Cypress tests. The app used these custom commands:

```ts
Cypress.Commands.add('nuxtLocalePath', () => {
  cy.window().then((win) => {
    return cy.wrap(win.$nuxt.localePath);
  });
});

Cypress.Commands.add('nuxtI18n', () => {
  cy.window().then((win) => {
    return cy.wrap(win.$nuxt.$i18n);
  });
});
```

Since `window.$nuxt` doesn't exist anymore in Nuxt 3, how could we manage to translate texts in this context ?

Thanks for the help !

**Comment by kosarsute** [comment]
1. Use the this.$nuxt syntax within the test
`Cypress.Commands.add('nuxtLocalePath', () => {
  return cy.wrap(this.$nuxt.localePath);
});`
2. Use the cy.window().its() syntax
`Cypress.Commands.add('nuxtLocalePath', () => {
  return cy.window().its('__NUXT__').its('data').its('state').its('$nuxt').its('localePath');
});`
Note that in both cases, you will need to ensure that the Nuxt application is loaded and running before running these commands. You can use the cy.visit() command to load the Nuxt application before running the tests.

I hope this helps!
